File: Problems/17-Split-a-String-in-Balanced-Strings/Split-a-String-in-Balanced-Strings.cpp
class Solution {
public:
    int balancedStringSplit(string s) {
        int ans = 0;
        int cpt = 0;
        for (auto& c: s)
        {
            if (c == 'R') cpt ++;
            else cpt --;

            if (cpt == 0) ans ++;
        }
        return ans;
    }
};

File: Problems/17-Split-a-String-in-Balanced-Strings/Split_a_String_in_Balanced_Strings.py
class Solution(object):
    def balancedStringSplit(self, s):
        """
        :type s: str
        :rtype: int
        """
        res, tmp = 0,0
        for char in s:
            if char == 'R':
                tmp += 1
            else:
                tmp -= 1
            if tmp == 0:
                res += 1
        return res